config.go

```diff
@@ -160,23 +120,74 @@ func makeCfg(config []Config) (cfg Config) {
 			cfg.AuthScheme = "Bearer"
 		}
 	}
-	if cfg.KeyRefreshTimeout == nil {
-		cfg.KeyRefreshTimeout = &defaultKeyRefreshTimeout
-	}
 
 	if cfg.KeyFunc == nil {
-		if len(cfg.KeySetURLs) > 0 {
-			jwks := &KeySet{
-				Config: &cfg,
+		if len(cfg.SigningKeys) > 0 || len(cfg.JWKSetURLs) > 0 {
+			var givenKeys map[string]keyfunc.GivenKey
+			if cfg.SigningKeys != nil {
+				givenKeys = make(map[string]keyfunc.GivenKey, len(cfg.SigningKeys))
+				for kid, key := range cfg.SigningKeys {
+					givenKeys[kid] = keyfunc.NewGivenCustom(key, keyfunc.GivenKeyOptions{}) // TODO User supplied alg?
+				}
+			}
+			if len(cfg.JWKSetURLs) > 0 {
+				var err error
+				if len(cfg.JWKSetURLs) == 1 {
+					cfg.KeyFunc, err = oneKeyfunc(cfg, givenKeys)
+				} else {
+					cfg.KeyFunc, err = multiKeyfunc(givenKeys, cfg.JWKSetURLs)
+				}
+				if err != nil {
+					panic("Failed to create keyfunc from JWK Set URL: " + err.Error()) // TODO Don't panic?
```

Comment:
There is more than one code path in this function that could panic. This panic could be the result of a failed HTTP request, as `keyfunc` may fail to create a JWK Set cache. Would changing this function to return an error be desired?

Reply:
Panic seems OK since it doesn't panic in the request. It panic when initting the middleware 

https://github.com/MicahParks/jwt/blob/master/main.go#L26

Reply:
I've removed the TODO.